### PR TITLE
Add translation support for nested strings

### PR DESCRIPTION
**Pull request by wackerow**
Fixes issue where intl variables used within `FormatMessage` component were not always set up for translation themselves.

example:
```ts
// Incorrect, as "not" does not get translated
              values={{ not: <em>not</em> }}
// Corrected to:
              values={{
                not: (
                  <em>
                    <FormattedMessage defaultMessage="not" />
                  </em>
                ),
              }}
```

- Extracts remaining strings, including "Mainnet" and "____ testnet" for translation.
- Also capitalized "Mainnet" and the testnet names for consistency

Note: There are some CLI variables such as `--num_validators` and `--chain` that are currently being translated. @CarlBeek Want to confirm that this is aligned with the `staking-deposit-cli` script.

### CODE REVIEW

File: src/components/AppBar.tsx [hunk 1]
```diff
@@ -151,9 +151,15 @@ const _AppBar = ({ location }: RouteComponentProps) => {
     ? TESTNET_LAUNCHPAD_URL
     : MAINNET_LAUNCHPAD_URL;
 
-  const networkName = IS_MAINNET
-    ? 'mainnet'
-    : `${EL_TESTNET_NAME.toLowerCase()} testnet`;
+  const networkName = IS_MAINNET ? (
```
Comment (samajammin) [inline src/components/AppBar.tsx:154]: Any particular reason we create a variable here vs. placing this inline with the HTML like we do elsewhere?

File: src/pages/TopUp/components/WalletConnectModal.tsx [hunk 2]
```diff
@@ -88,11 +88,16 @@ const WalletConnectModal: React.FC<{
             <FormattedMessage
               defaultMessage="Connect to {network}"
               values={{
-                network: IS_MAINNET
-                  ? 'Ethereum mainnet'
-                  : `${EL_TESTNET_NAME.toLowerCase()} testnet`,
+                network: IS_MAINNET ? (
```
Comment (samajammin) [inline src/pages/TopUp/components/WalletConnectModal.tsx:91]: Seems like we duplicate this exact same logic in a few places - I wonder if we could make this more DRY?

**Review by samajammin — APPROVED** [review]
A couple minor suggestions - overall looks good!

File: src/pages/TopUp/components/WalletConnectModal.tsx (hunk 2 above)
Comment (wackerow) [reply src/pages/TopUp/components/WalletConnectModal.tsx]: Agree with this sentiment... Just added some commits implementing a `useIntlNetworkName` hook to remove a lot of the redundancy here.

Can access using `const { executionLayerName, consensusLayerName } = useIntlNetworkName();`

File: src/hooks/useIntlNetworkName.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { useIntl } from 'react-intl';
+import {
+  IS_MAINNET,
+  EL_TESTNET_NAME,
+  TESTNET_LAUNCHPAD_NAME,
+} from '../utils/envVars';
+
+const useIntlNetworkName = (): {
```
Comment (samajammin) [inline src/hooks/useIntlNetworkName.tsx:8]: 🔥

**Comment by samajammin** [comment]
I thought the code looks good but something appears to be screwy with the logic:
Image 2022-07-27 at 11 07 15 AM
- lowercase `mainnet` should be uppercase
- Either the current networks is wrong or the `Switch to Mainnet` is wrong

**Comment by samajammin** [comment]
I'm still confused. Why would we use `NETWORK_NAME` = `'Mainnet'` if `IS_MAINNET` is `false`?

Perhaps we should update:
```
export const NETWORK_NAME          = (IS_MAINNET || !process.env.REACT_APP_TESTNET_LAUNCHPAD_NAME) ? 'Mainnet' : process.env.REACT_APP_TESTNET_LAUNCHPAD_NAME;
```
to be
```
export const NETWORK_NAME          = IS_MAINNET ? 'Mainnet' : TESTNET_LAUNCHPAD_NAME;
```
Given that we have a fallback for that variable if `REACT_APP_TESTNET_LAUNCHPAD_NAME` is falsey?

**Review by samajammin — APPROVED** [review]
👍

**Pull request merged**